R/determine_dp.R

```diff
@@ -0,0 +1,57 @@
+#' Determine decimal places for large numeric values
+#'
+#' This helper function calculates the appropriate number of decimal places
+#' based on the value's magnitude. Values smaller than 1 million use the
+#' supplied default decimal places. For values over 1 million or 1 billion,
+#' decimal places are conditionally applied if the value normalised by a
+#' million or billion is not divisible by 10.
+#'
+#' @param value A single numeric value.
+#' @param dp Integer. The default number of decimal places for values
+#' over 1 million or 1 billion.
+#' @param dynamic_dp_value Integer. Default is 2. Sets the number of decimal
+#' places to use when the value is ≥ 1 million or ≥ 1 billion but not
+#' divisible by 10 after scaling. This adds precision only when needed,
+#' improving clarity without over-formatting fo pretty_num().
+#' @return An integer indicating the number of decimal places to use.
+#' @keywords internal
+#' @export
+#' @examples
+#' determine_dp(999999) # Returns 0 (less than 1 million)
+#' determine_dp(1e6) # Returns 2 (not divisible by 10 after scaling)
+#' determine_dp(10e6) # Returns 0 (divisible by 10 after scaling)
+#' determine_dp(1e9) # Returns 2 (not divisible by 10 after scaling)
+#' determine_dp(10e9) # Returns 0 (divisible by 10 after scaling)
+#' determine_dp(-1e6) # Returns 2 (absolute value logic)
+#' determine_dp(1e6, dp = 0, dynamic_dp_value = 4) # Returns 4
+#' determine_dp(500000, dp = 1, dynamic_dp_value = 3) # Returns 1
+#' determine_dp(10e6, dp = 1, dynamic_dp_value = 3) # Returns 0
+#' determine_dp(-2e6, dp = 3, dynamic_dp_value = 5) # Returns 5
```

Comment:
You don't need to add the expected results in comments after each line - in the main docs these will be produced and printed underneath each line, e.g. 

<img width="736" height="779" alt="Image" src="https://github.com/user-attachments/assets/d02d4c8b-8e89-4c1d-8536-e272f29f4858" />